torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -2316,6 +2454,281 @@ size_t expandVectorizationToContigMergedDomains(
+              [&exact_disjoint_set](
+                  std::pair<IterDomain*, IterDomain*> id_pair) {
+                return !exact_disjoint_set.strictAreMapped(
+                    id_pair.first, id_pair.second);
+              })) {
+        return false;
+      }
+    }
+    {
+      std::vector<std::pair<IterDomain*, IterDomain*>> zipped_ids;
+
+      std::transform(
+          first_rfactor_dom.begin(),
+          first_rfactor_dom.end(),
+          other_rfactor_dom.begin(),
+          std::back_inserter(zipped_ids),
+          [](IterDomain* first, IterDomain* second) {
+            return std::make_pair(first, second);
+          });
+
+      if (std::any_of(
+              zipped_ids.begin(),
+              zipped_ids.end(),
+              [&exact_disjoint_set](
+                  std::pair<IterDomain*, IterDomain*> id_pair) {
+                return !exact_disjoint_set.strictAreMapped(
+                    id_pair.first, id_pair.second);
+              })) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+class DependentViewCheck : public IterVisitor {
+ private:
+  bool dep_views_ = false;
+  // All values that are dependent on a view expression
+  std::unordered_set<Val*> dep_on_view_;
+
+  std::vector<Statement*> next(Val* v) final {
+    if (dep_views_) {
+      return {};
+    }
+    return IterVisitor::next(v);
+  }
+
+  std::vector<Statement*> next(Expr* expr) final {
+    if (dep_views_) {
+      return {};
+    }
+    return IterVisitor::next(expr);
+  }
+
+  void handle(Expr* e) final {
+    // Check if any inputs have a dependency on a view operation
+    bool has_view_dep_input =
+        std::any_of(e->inputs().begin(), e->inputs().end(), [&](Val* v) {
+          return dep_on_view_.count(v);
+        });
+
+    auto out_tvs = ir_utils::filterByType<TensorView>(e->outputs());
+
+    bool is_view_op = e->isA<ViewOp>() &&
+        std::any_of(out_tvs.begin(), out_tvs.end(), [](TensorView* out_tv) {
+                        return out_tv->hasRFactor();
+                      });
+
+    // Forward dependencies on views, assume if any input is dependent on a
+    // view all outputs are (not necessarily true with grouped reductions.)
+    if (has_view_dep_input || is_view_op) {
+      dep_on_view_.insert(e->outputs().begin(), e->outputs().end());
+    }
+
+    if (has_view_dep_input && is_view_op) {
+      // Inputs are dependent on a view, and this operation is a view so
+      // there's views that are dependent on views.
+      dep_views_ = true;
+    }
+  }
+
+ public:
+  // Return if there are any view's that have dependencies on other view
+  // operations.
+  static bool hasDependentViews(Fusion* fusion) {
+    DependentViewCheck view_check;
+    view_check.traverse(fusion);
+    return view_check.dep_views_;
+  }
+};
+
+bool hasDependentViews(Fusion* fusion) {
+  return DependentViewCheck::hasDependentViews(fusion);
+}
+
+TORCH_CUDA_CU_API bool breakIsDisjoint(std::vector<int> group_ids, int pos) {
```

Comment:
nit: We can omit the `TORCH_CUDA_CU_API` in a cpp file
```suggestion
bool breakIsDisjoint(std::vector<int> group_ids, int pos) {
```